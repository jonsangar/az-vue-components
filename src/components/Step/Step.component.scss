$color-background: var(--color-background);

.az-step{
  position: relative;
  color: color(gray5);

  &--active{
    color: color(gray2);

    .az-step__title {
      color: color(gray2);
    }
  }

  &--success{
    color: color(gray2);

    .az-step__title {
      color: color(success);
    }
  }

  &--error{
    color: color(danger);

    .az-step__title {
      color: color(danger);
    }
  }

  &__title{
    text-align: center;
  }

  &__icon{
    position: relative;
    z-index: 1;
    display: inline-block;
    width: 100%;
    padding: 0 half();
    text-align: center;
    background-color: $color-background;
  }

  // &__line{
  //   z-index: 0;
  //   height: 1px;
  //   background-color: color(gray5);
  //   transform: translateY(-35px);
  // }


}


// $color-circle: #ffffff;
// $color-line: color(gray5);
// $animation-duration-base: 1s;
// $step-horizontal-title-font-size: font-size(small1);
// $step-icon-size: 29px;
// $step-process-text-color: green;
// $step-finish-text-color: blue;
// $step-active-color: orange;
// $step-finish-line-color: gray;

// .az-step {
//   position: relative;
//   flex: 1;
//   font-size: font-size(small1);
//   color: color(gray4);

//   &__circle {
//     display: block;
//     width: 50px;
//     height: 50px;
//     background-color: $color-circle;
//     border-radius: 50%;
//   }

//   &__line {
//     position: absolute;
//     background-color: $color-line;
//     transition: background-color $animation-duration-base;
//   }

//   &--horizontal {
//     float: left;

//     &:first-child {
//       .az-step__title {
//         margin-left: 0;
//         transform: none;
//       }
//     }

//     &:last-child {
//       position: absolute;
//       right: 1px;
//       width: auto;

//       .az-step__title {
//         margin-left: 0;
//         transform: none;
//       }

//       .az-step__circle-container {
//         right: -9px;
//         left: auto;
//       }
//     }

//     .az-step__circle-container {
//       position: absolute;
//       top: 30px;
//       left: simple();
//       z-index: 1;
//       padding: 0 simple();
//       background-color: #ffffff;
//       transform: translateY(-50%);
//     }

//     .az-step__title {
//       display: inline-block;
//       margin-left: 3px;
//       font-size: $step-horizontal-title-font-size;
//       transform: translateX(-50%);

//       @media (max-width: 321px) {
//         font-size: $step-horizontal-title-font-size - 1px;
//       }
//     }

//     .az-step__line {
//       top: 30px;
//       left: 0;
//       width: 100%;
//       height: 1px;
//     }

//     .az-step__icon {
//       display: block;
//       font-size: $step-icon-size;
//     }

//     .az-step--process {
//       color: $step-process-text-color;
//     }
//   }

//   &--vertical {
//     display: block;
//     float: none;
//     padding: 10px 10px 10px 0;
//     line-height: simple();

//     &:not(:last-child)::after {
//       border-bottom-width: 1px;
//     }

//     &:first-child {
//       &::before {
//         position: absolute;
//         top: 0;
//         left: -15px;
//         z-index: 1;
//         width: 1px;
//         height: 20px;
//         content: '';
//         background-color: #ffffff;
//       }
//     }

//     .az-step__circle-container {
//       position: absolute;
//       top: 19px;
//       left: -15px;
//       z-index: 2;
//       font-size: $step-icon-size;
//       line-height: 1;
//       transform: translate(-50%, -50%);
//     }

//     .az-step__line {
//       top: 16px;
//       left: -15px;
//       width: 1px;
//       height: 100%;
//     }
//   }

//   &:last-child {
//     .az-step__line {
//       width: 0;
//     }
//   }

//   &--finish {
//     color: $step-finish-text-color;

//     .az-step__circle,
//     .az-step__line {
//       background-color: $step-finish-line-color;
//     }
//   }

//   &__icon,
//   &__title {
//     transition: color $animation-duration-base;

//     &--active {
//       color: $step-active-color;
//     }
//   }
// }
