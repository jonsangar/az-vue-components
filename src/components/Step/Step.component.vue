<template lang="pug">

.az-step(
  :class="{'az-step--active': isActive, 'az-step--error': hasError, 'az-step--success': isSuccess}")
  .az-step__icon
    az-icon(:name="setIcon" scale="1")
  .az-step__title(
    :class="'az-step__title--active'")
      <slot v-if="title === 'Step'">Step {{ order }}</slot>
      <slot v-else>{{ title }}</slot>
  //- .az-step__line
</template>

<script>
export default {

  props: {
    title: {
      type: String,
      default: 'Step'
    },

    order: {
      type: [Number,String],
      required: false
    },

    isActive: {
      type: Boolean,
      default: false
    },

    hasError: {
      type: Boolean,
      default: false
    },

    isSuccess: {
      type: Boolean,
      default: false,
    },

    icon: {
      type: String,
      default: 'save'
    }
  },

  computed: {
    iconDefault() {
      return (this.isActive && this.icon === 'save') ? 'save' : 'cancel'
    },

    setIcon() {
      const icon = this.iconDefault;
      if(this.icon === 'save') {
        return icon
      }

      return this.icon;
    }
  }

}
</script>

<style lang="scss" src="./Step.component.scss" scoped></style>
