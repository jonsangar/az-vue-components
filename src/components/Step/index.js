import Step from './Step.component.vue';

export default Step;
