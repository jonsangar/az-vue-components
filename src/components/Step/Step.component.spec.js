import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerÃ­as de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Step from './Step.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Step.vue', () => {
  const build = () => {
    const wrapper = shallowMount(Step, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Step con props por defecto con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.vm.title).toBe('Step')
    expect(wrapper.vm.isActive).toBe(false)
    expect(wrapper.vm.hasError).toBe(false)
    expect(wrapper.vm.isSuccess).toBe(false)
    expect(wrapper.vm.icon).toBe('save')
  })

  it('Comprueba que se actualizan las props cuando se asignan las props', async () => {
    const { wrapper } = build()
    wrapper.setProps({
      title: 'MyStep',
      order: '1',
      isActive: true,
      isSuccess: true,
      icon: 'check',
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.title).toBe('MyStep')
    expect(wrapper.vm.order).toBe('1')
    expect(wrapper.vm.isActive).toBe(true)
    expect(wrapper.vm.isSuccess).toBe(true)
    expect(wrapper.vm.icon).toBe('check')
  });
});
