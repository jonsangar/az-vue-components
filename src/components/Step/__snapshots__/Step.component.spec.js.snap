// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step.vue Comprueba renderizado del componente Step con props por defecto con snapshot 1`] = `
<div class="az-step">
  <div class="az-step__icon">
    <az-icon-stub name="cancel" width="" height="" scale="1" fill="true"></az-icon-stub>
  </div>
  <div class="az-step__title az-step__title--active">Step </div>
</div>
`;
